h5netcdf/core.py

```diff
@@ -253,22 +279,68 @@ def __init__(self, parent, name):
 
                     self._dim_sizes[k] = size
+
+                    # keep track of found labeled dimensions
+                    if self._root._phony_dim_count is not None:
+                        labeled_dims[size] += 1
+                        self._phony_dims[(size, labeled_dims[size] - 1)] = k
+
                     # Figure out the current size of a dimension, which for
                     # unlimited dimensions requires looking at the actual
                     # variables.
                     self._current_dim_sizes[k] = \
                         self._determine_current_dimension_size(k, current_size)
 
                     self._dim_order[k] = dim_id
+                else:
+                    if self._root._phony_dim_count is not None:
+                        # check if malformed variable
+                        if not _unlabeled_dimension_mix(v):
+                            # if unscaled variable, get phony dimensions
+                            vdims = defaultdict(int)
+                            for i in v.shape:
+                                vdims[i] += 1
+                            for dimsize, cnt in vdims.items():
+                                phony_dims[dimsize] = max(phony_dims[dimsize],
+                                                          cnt)
+
                 if not _netcdf_dimension_but_not_variable(v):
                     if isinstance(v, h5py.Dataset):
                         var_name = k
                         if k.startswith('_nc4_non_coord_'):
                             var_name = k[len('_nc4_non_coord_'):]
                         self._variables.add(var_name)
 
+        # iterate over found phony dimensions and create them
+        if self._root._phony_dim_count is not None:
+            grp_phony_count = 0
+            for size, cnt in phony_dims.items():
+                # only create missing dimensions
+                for pcnt in range(labeled_dims[size], cnt):
+                    name = (grp_phony_count +
+                            self._root._phony_dim_count)
+                    grp_phony_count += 1
+                    # this is for per access naming
+                    if self._root._labeled_dim_count is None:
+                        name = "phony_dim_{}".format(name)
+                        self._create_dimension(name, size)
+                    self._phony_dims[(size, pcnt)] = name
+            # finally increase phony dim count at file level
+            self._root._phony_dim_count += grp_phony_count
+
         self._initialized = True
 
+    def _create_phony_dimensions(self):
+        # this is for 'sort' naming
+        for key, value in self._phony_dims.items():
+            try:
+                value += self._root._labeled_dim_count
+                name = "phony_dim_{}".format(value)
+                self._create_dimension(name, key[0])
+                self._phony_dims[key] = name
+            except TypeError:
+                pass
```

Comment:
This is a rather broad exception to be catching over multiple lines of code that could hide programming errors. When can this arise?

Reply:
I could check the `value` with isinstance(int, value) which would have the same effect. Would that be OK?

Reply:
Yes, that would be much better!